tools/check-signed-off.sh

```diff
@@ -28,16 +69,27 @@ else
  exit 1
 fi
 
-author_name=`git show -s --format=%an $commit_hash`
-author_email=`git show -s --format=%ae $commit_hash`
-required_signed_off_by_line="JerryScript-DCO-1.0-Signed-off-by: $author_name $author_email"
+# Checking the last line
 actual_signed_off_by_line=`git show -s --format=%B $commit_hash | sed '/^$/d' | tr -d '\015' | tail -n 1`
 
-if [ "$actual_signed_off_by_line" != "$required_signed_off_by_line" ]
+if [ "$TOLERANT" == "no" ]
 then
- echo -e "\e[1;33mSigned-off-by message is incorrect. The following line should be at the end of the $commit_hash commit's message: '$required_signed_off_by_line'. \e[0m"
+ author_name=`git show -s --format=%an $commit_hash`
+ author_email=`git show -s --format=%ae $commit_hash`
+ required_signed_off_by_line="JerryScript-DCO-1.0-Signed-off-by: $author_name $author_email"
 
- exit 1
+ if [ "$actual_signed_off_by_line" != "$required_signed_off_by_line" ]
+ then
+  echo -e "\e[1;33mSigned-off-by message is incorrect. The following line should be at the end of the $commit_hash commit's message: '$required_signed_off_by_line'. \e[0m"
+  exit 1
+ fi
+else
+  echo -e "\e[1;33mWarning! The name and email address of the author of the $commit_hash commit is not checked in tolerant mode! \e[0m"
+  if echo "$actual_signed_off_by_line" | grep -q -v '^JerryScript-DCO-1.0-Signed-off-by:'
```

Comment:
I think this check should be a RegEx that checks the format after the colon (`firstname lastname [...] email@domain`). But that might cause similar problems to strict mode.